doc/nrf/protocols/matter/getting_started/memory_profiling.rst

```diff
@@ -0,0 +1,371 @@
+.. _ug_matter_device_memory_profiling:
+
+Profiling memory in Matter applications
+#######################################
+
+.. contents::
+   :local:
+   :depth: 2
+
+This guide offers useful commands and Kconfig options for configuring memory sizes for stacks, heaps, and Zephyr settings.
```

Comment:
This sounds a little bit like you could configure memory size using commands. How about the following?
```suggestion
This guide offers useful commands for measuring usage of stacks, heaps, and Zephyr settings, and Kconfig options for configuring their memory size.
```